packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/ArbTokenBridge.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright 2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pragma solidity ^0.6.11;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "./StandardArbERC20.sol";
+import "./StandardArbERC777.sol";
+import "arb-bridge-eth/contracts/libraries/CloneFactory.sol";
+
+import "./IArbToken.sol";
+import "arb-bridge-eth/contracts/libraries/ICloneable.sol";
+import "arbos-contracts/contracts/ArbSys.sol";
+
+contract ArbTokenBridge is CloneFactory {
+    using Address for address;
+
+    /// @notice This mapping is from L1 address to L2 address
+    mapping(address => address) public customToken;
+
+    uint256 exitNum;
+
+    ICloneable public immutable templateERC20;
+    ICloneable public immutable templateERC777;
+
+    modifier onlyEthPair {
+        // This ensures that this method can only be called from the L1 pair of this contract
+        require(tx.origin == address(this), "ONLY_ETH_PAIR");
+        _;
+    }
+
+    constructor() public {
+        templateERC20 = new StandardArbERC20();
+        templateERC777 = new StandardArbERC777();
+    }
+
+    function mintFromL1(
```

Comment:
I think we should be explicit on the calls that use the ERC777 standard, the same way as ERC20. Adding the standard to each function name helps reduce possible error, so users are more explicit about what they are using.

My main concern is that someone may interact with this function without knowing this is using a standard with callback hooks under the hood.